File: .cpp/Program/Smartphone.cpp
#pragma once
#include <bits/stdc++.h>
#include "Computer.cpp"
using namespace std;

class Smartphone : public Computer {
    private:
        int batteryCapacity;
        int cameraResolution;     

    public:
        Smartphone () { }

        Smartphone (string brand, string model, string OS, CPU cpu, vector<RAM> ram, vector<Storage*> storage, int batteryCapacity, int cameraResolution) : Computer (brand, model, OS, cpu, ram, storage) {
            this->batteryCapacity = batteryCapacity;
            this->cameraResolution = cameraResolution;            
        }

        void setBatteryCapacity (int batteryCapacity) { this->batteryCapacity = batteryCapacity; }
        void setCameraResolution (int cameraResolution) { this->cameraResolution = cameraResolution; }

        int getBatteryCapacity () { return batteryCapacity; }
        int getCameraResolution () { return cameraResolution; }

        void info () override {
            cout << "Smartphone" << endl;
            Computer::info();
            cout << "| Battery Capacity  : " << batteryCapacity << "mAh" << endl;
            cout << "| Camera Resolution : " << cameraResolution << "MP" << endl;
        }

        ~Smartphone () { }
};

File: .py/Program/Smartphone.py
from Computer import Computer

class Smartphone (Computer):
    __batteryCapacity = 0
    __cameraResolution = 0

    def __init__ (self, brand = "", model = "", OS = "", CPU = None, RAM = None, Storage = None, batteryCapacity = 0, cameraResolution = 0):
        super ().__init__ (brand, model, OS, CPU, RAM, Storage)
        self.__batteryCapacity = batteryCapacity
        self.__cameraResolution = cameraResolution

    def setBatteryCapacity (self, batteryCapacity):
        self.__batteryCapacity = batteryCapacity

    def setCameraResolution (self, cameraResolution):
        self.__cameraResolution = cameraResolution

    def getBatteryCapacity (self):
        return self.__batteryCapacity
    
    def getCameraResolution (self):
        return self.__cameraResolution
    
    def info (self):
        print ("| Smartphone")
        super ().info ()
        print (f"| Battery Capacity  : { self.__batteryCapacity }mAh")
        print (f"| Camera Resolution : { self.__cameraResolution }MP")